crates/ra_hir/src/ty.rs

```diff
@@ -449,6 +449,12 @@ pub fn type_for_enum(db: &impl HirDatabase, s: Enum) -> Cancelable<Ty> {
     })
 }
 
+pub fn type_for_enum_variant(db: &impl HirDatabase, s: EnumVariant) -> Cancelable<Ty> {
+    // TODO: need some way to get at the enum
+    let e: Enum = unimplemented!();
```

Comment:
yeah, I think we'll want a method on `EnumVariant` that returns the `Enum`. I think it makes sense to store the def_id of the containing enum in the `EnumVariantData`; you should be able to go from the `DefLoc` of the variant to the `DefLoc` of the enum by searching for the `ast::EnumDef` in the ancestors and then get the `DefId` for that again. Though maybe @matklad has a better suggestion :)

Reply:
Sensible - I'll try to make that work.